indico/modules/events/timetable/client/js/dnd/dnd.tsx

```diff
@@ -258,33 +261,29 @@ export function DnDProvider({
     setDraggables(d => removeKey(d, id));
   }, []);
 
-  const onMouseDown = useCallback(
-    (id, {x, y, offsetX, offsetY}) => {
-      if (state.current.state === 'idle') {
-        const draggable = draggables[id];
-        if (!draggable) {
-          return;
-        }
-
-        const scrollParent = getScrollParent(draggable.node.current); // TODO: this should be getTotalScroll()
-        state.current = {
-          state: 'mousedown',
-          initialMousePosition: {x, y},
-          scrollPosition: {x: 0, y: 0},
-          initialScrollPosition: {x: scrollParent.scrollLeft, y: scrollParent.scrollTop},
-          initialOffset: {x: offsetX, y: offsetY},
-          activeDraggable: id,
-        };
-        setDraggableData(d =>
-          setInitialOffset(setBoundingRectAndScroll(d, draggable.node, id), id, {
-            x: offsetX,
-            y: offsetY,
-          })
-        );
+  const onMouseDown = useCallback((id: string, draggable: Draggable, {x, y, offsetX, offsetY}) => {
+    if (state.current.state === 'idle') {
+      if (!draggable) {
+        return;
```

Comment:
Is this check still necessary now? Do we pass empty draggable?

Reply:
Good question, I think it might be possible to call `unregisterDraggable` and then get undefined for a draggable from the context selector inside the useDraggable hook. Unfortunately the types don't really reflect that currently, probably something to improve eventually as well.

Reply:
Added a todo comment ^